### PR TITLE
Issue #10: clear cached call upon a drupal cache clear

### PR DESCRIPTION
**Pull request by jaapjansma**

### CODE REVIEW

**Comment by jaapjansma** [comment]
@systopia could you review this pr?

File: CMRF_Drupal/SQLPersistingCallFactory.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  /** @var mysqli */
+  private $connection;
+
+  public function __construct(mysqli $sql_connection, $table_name, callable $constructor, callable $loader) {
+    parent::__construct($sql_connection, $table_name, $constructor, $loader);
+    $this->connection=$sql_connection;
+  }
+  
+  public function clearCachedCalls() {
+    $stmt = $this->connection->query("delete from {$this->table_name} where status = 'DONE'");
```
Comment (jensschuppe) [inline CMRF_Drupal/SQLPersistingCallFactory.php]: `$this->table_name` has private access in base class `\CMRF\PersistenceLayer\SQLPersistingCallFactory,` should be protected for `SQLPersistingCallFactory` to be able to access.

Maybe all private members and methods of `\CMRF\PersistenceLayer\SQLPersistingCallFactory` should be protected?

File: cmrf_core.module [hunk 2]
```diff
@@ -184,6 +184,16 @@ function cmrf_core_cron() {
   }
 }
 
+function cmrf_core_flush_caches() {
```
Comment (jensschuppe) [inline cmrf_core.module]: This should include the docblock for implemented hooks:
```
/**
* Implements hook_flush_caches().
*/
```

**Review by jensschuppe — CHANGES_REQUESTED** [review]

File: CMRF_Drupal/SQLPersistingCallFactory.php (hunk 1 above)
Comment (jaapjansma) [reply CMRF_Drupal/SQLPersistingCallFactory.php]: You are right. That should be fixed in the abstract core.

**Comment by jaapjansma** [comment]
@jensschuppe thanks for your review. I have pushed to the abstract core to make the property of the SQLPersistanceCallFactory protected. I have also added a doc block. Could you review it again?

**Review by jensschuppe — APPROVED** [review]

**Pull request merged**